runtime/src/test/java/com/flipkart/flux/dao/StateMachinesDAOTest.java

```diff
@@ -13,56 +13,87 @@
 
 package com.flipkart.flux.dao;
 
-import java.util.HashSet;
-import java.util.Set;
-
-import com.flipkart.flux.dao.iface.StatesDAO;
-import org.junit.Before;
-import org.junit.Test;
-
+import com.flipkart.flux.api.EventDefinition;
+import com.flipkart.flux.api.StateDefinition;
+import com.flipkart.flux.api.StateMachineDefinition;
 import com.flipkart.flux.dao.iface.StateMachinesDAO;
+import com.flipkart.flux.dao.iface.StatesDAO;
 import com.flipkart.flux.domain.State;
 import com.flipkart.flux.domain.StateMachine;
-import com.flipkart.flux.guice.module.ConfigModule;
-import com.flipkart.flux.guice.module.HibernateModule;
-import com.google.inject.Guice;
-import com.google.inject.Injector;
-
+import com.flipkart.flux.representation.DomainTypeCreator;
+import com.flipkart.flux.rules.DbClearRule;
+import com.flipkart.flux.runner.GuiceJunit4Runner;
 import junit.framework.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
 
+import javax.inject.Inject;
 import javax.transaction.Transactional;
+import java.util.HashSet;
+import java.util.Set;
 
 /**
  * <code>StateMachinesDAOTest</code> class tests the functionality of {@link StateMachinesDAO} using JUnit tests.
  * @author shyam.akirala
  * @author kartik.bommepally
  */
+@RunWith(GuiceJunit4Runner.class)
 public class StateMachinesDAOTest {
 
-    private Injector injector;
+    @Inject
+    @Rule
+    public DbClearRule dbClearRule;
+
+    @Inject
+    StateMachinesDAO stateMachinesDAO;
+
+    @Inject
+    StatesDAO statesDAO;
+
+    @Inject
+    DomainTypeCreator domainTypeCreator;
 
     @Before
-    public void setup() {
-        injector = Guice.createInjector(new ConfigModule(), new HibernateModule());
-    }
+    public void setup() {}
 
-    @Test
-    @Transactional
+    @Test @Transactional
```

Comment:
Thanks for the point. I thought it's getting intercepted, but it's not as Guice is not creating it's instance. @yogeshnachnani